CHANGELOG.md

```diff
@@ -29,6 +29,10 @@ _ `_optional` subpackage for managing optional dependencies
 - `register_hooks` utility enabling user-defined augmentation of arbitrary callables
 - `transform` methods of `SearchSpace`, `SubspaceDiscrete` and `SubspaceContinuous`
   now take additional `allow_missing` and `allow_extra` keyword arguments
+- `SubspaceDiscrete.from_parameter`, `SubspaceContinuous.from_parameter` and
+  `SearchSpace.from_parameter` convenience constructors
+- `DiscreteParameter.to_subspace`, `ContinuousParameter.to_subspace` and
```

Comment:
I have to say that I'm struggling to see the benefit here. Of course, having more convenience constructors is nice, but is this really necessary at the moment? In particular, the case with users having search spaces with a single parameter only is something that does not even make sense to me. Have there been reports about users wanting this feature/having trouble with this?

Reply:
What, why doesn't it make sense to you? Having a single parameter is a perfectly fine use case. Where do you see the problem?

Well, it's not super critical, but it has been bothering me enough in the last months that I decided to invest these 30mins to finally fix it. When I develop, I regularly test different different configurations (apart from our existing tests) and often build search spaces on the fly. The `from_product(parameters=[param])` pattern has annoyed me ever since – it's exactly the same thing that we originally had with our "Objective" where we had to type `Objective(targets=[target])`, which I am very happy about that we got this clean now.

But apart from my personal frustration, how would you justify to a user who wants to create a 1-d searchspace they have to go via a product creation step that does not involve creating a product?!

Reply:
Well, I just struggle to see a reasonable 1-Dim use Case for Bayesian Optimization but this might just be my limited imagination 😄 This however raises the question: Why do you have additional tests on your own and why are they not part of our pipeline?

Reply:
Not really "tests" in the pytest meaning. It's for me when I draft code, I create temporary code to play around with (basically: developing code in the debugger) and create objects on the fly to test different aspects of newly written code / ideas. 

But coming back to the original discussion: do you see the analogy with the `SingleTargetObjective` and can do agree that is simply fills a missing piece in our constructor logic? Or how shall we continue now?

Reply:
I see the analogy with the objective and I am fine with continuing here :)

Reply:
Alright, then I'll resolve. Thx 👍🏼